tomcat/tomcat-jta/src/test/java/org/jboss/narayana/tomcat/jta/integration/AbstractCase.java

```diff
@@ -0,0 +1,191 @@
+ *
+ * @author <a href="mailto:karm@redhat.com">Michal Karm Babacek</a>
+ */
+public abstract class AbstractCase {
+
+    private static final Logger LOGGER = Logger.getLogger(AbstractCase.class.getName());
+
+    /**
+     * The deployment is the same for each test, but context.xml and DB driver jar
+     * changes according to the database used.
+     */
+    static WebArchive webArchive;
+    static Allocator dba;
+    static DB db;
+    private static File[] libFiles;
+    static String catalinaHome;
+
+    /**
+     * Acquire resources and prepare configuration
+     */
+    @BeforeClass
+    public static void init() {
+        try {
+            dba = Allocator.getInstance();
+            LOGGER.info("Allocating a new database might take many minutes, depending on the mode the test suite operates in.");
+            db = dba.allocateDB();
+            assertNotNull("Failed to allocate DB. Check logs for the root cause.", db);
+            // Configuration of deployment, driver, data sources XML, ...
+            prepareContextXML();
+            final File[] dbDriver = (dba instanceof DBAllocator) ? new File[]{new File(db.dbDriverArtifact)} :
+                    Maven.resolver().resolve(db.dbDriverArtifact).withTransitivity().asFile();
+            assertNotNull("WebArchive was not created by @Deployment before @BeforeClass. Arquillian lifecycle config error?", webArchive);
+            webArchive.addAsLibraries(dbDriver);
+            webArchive.addAsManifestResource("context.xml", "context.xml");
+
+            final String versionTXSpec = System.getProperty("version.org.jboss.spec.javax.transaction");
+            if (StringUtils.isEmpty(versionTXSpec)) {
+                throw new IllegalArgumentException("version.org.jboss.spec.javax.transaction must not be empty");
+            }
+            catalinaHome = System.getenv("CATALINA_HOME");
+            if (StringUtils.isEmpty(catalinaHome)) {
+                throw new IllegalArgumentException("CATALINA_HOME must not be empty");
+            }
+            // @see https://issues.jboss.org/browse/JWS-976
+            libFiles = Maven.resolver()
+                    .resolve("org.jboss.spec.javax.transaction:jboss-transaction-api_1.2_spec:" + versionTXSpec)
+                    .withTransitivity().asFile();
+            final String dest = catalinaHome + File.separator + "lib" + File.separator + libFiles[0].getName();
+            try {
+                Files.copy(Paths.get(libFiles[0].toURI()),
+                        Paths.get(dest),
+                        StandardCopyOption.REPLACE_EXISTING);
+            } catch (IOException e) {
+                fail("Either CATALINA_HOME or version.org.jboss.spec.javax.transaction is invalid, we cannot copy to " + dest);
+                e.printStackTrace();
+            }
+        } catch (IOException | NullPointerException e) {
+            e.printStackTrace();
+            fail(e.getMessage());
+            if (dba != null && db != null) {
```

Comment:
place under `finally` block, this will be never executed (`fail` would throw `AssertionException` I expect)